### DISCUSSION TITLE
What is the recommended way to generate `AssetMaterialization` events for source assets managed outside of Dagster?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#14951] [Q&A] [answered]
Hello all, quick question. Is there a way to log an `AssetMaterialization` event from a sensor? I know how to log the materialization event from an `op`  but I’m wondering if I can do the same from a sensor for a `SourceAsset` managed outside Dagster.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1687790386681569?thread_ts=1687790386.681569&cid=C01U954MEER).*

**Comment by smackesey** [comment, marked as answer]
There is currently no public API for logging an `AssetMaterialization` event from a sensor. The recommended way to do this is to instead use a job on a schedule and yield the `AssetMaterialization` events from an op in the job.

cc @sryza for confirmation

**Reply by sryza** [reply]
Confirmed. If directly logging `AssetMaterialization`s from sensors is desired, please file an issue. Also, observable source assets are an approach that can be useful in this setting, instead of sensors: https://docs.dagster.io/concepts/assets/asset-observations#observable-source-assets.
